xpsi/PostProcessing/_cache.py

```diff
@@ -91,7 +91,7 @@ def __next__(self):
 
         with self._open('r') as f:
             g = f['data']
-            for key in g.keys():
+            for key in list(g.keys()):
```

Comment:
I don't think you need to turn this into a list here, do you? `g.keys()` should be a generator that you can step through in a for-loop? Might be worth checking.